Kudu.Core/Infrastructure/ProcessExtensions.cs

```diff
@@ -244,7 +244,14 @@ public static string GetCommandLine(this Process process)
 
         public static bool GetIsScmSite(Dictionary<string, string> environment)
         {
-            return environment[WellKnownEnvironmentVariables.ApplicationPoolId].StartsWith("~1", StringComparison.OrdinalIgnoreCase);
+            string appPool = null;
+            if (environment.TryGetValue(WellKnownEnvironmentVariables.ApplicationPoolId, out appPool) &&
+                !string.IsNullOrEmpty(appPool))
```

Comment:
The dictionary can contain a key, with a null value, and TryGetValue will return it and return true. I'm guarding against that.